framework/doc/content/syntax/MFEM/index.md

```diff
@@ -0,0 +1,90 @@
+
+!listing test/tests/mfem/kernels/diffusion.i block=/Mesh
+
+Then, we must set up the finite element spaces our problem will make use of. They can be of [MFEMScalarFESpace.md] type, which allows for continuous `H1` or discontinuous `L2` elements, or [MFEMVectorFESpace.md] type, thereby allowing tangentially-continuous `ND` or normally-continuous `RT` elements.
+
+!listing test/tests/mfem/kernels/diffusion.i block=/FESpaces
+
+In this diffusion example, besides the usual `H1` space required to solve the system, we also include an `ND` space so that we may output the gradient of the result as an `AuxVariable` later on. However, if you only wish to visualize the scalar result, it is not necessary to include the `ND` space.
+
+### Equation System - Variables, Kernels, and Boundary Conditions
+
+Having created the necessary finite element spaces, we may now set up the variables to be solved for. They should be of type [MFEMVariable.md]. Each variable should also be associated with a relevant finite element space.
+
+!listing test/tests/mfem/kernels/diffusion.i block=/Variables
+
+Here, if we also wish to output the gradient of the result, we can add it as an `AuxVariable` and set up its corresponding [`AuxKernel`](MFEMAuxKernel.md).
+
+!listing test/tests/mfem/kernels/diffusion.i block=/AuxVariables AuxKernels
+
+Then, within the `Kernels` block, we specify the weak forms to be added to our equation system. Typically, one would pick the MFEM integrators they wish to implement by checking the [linear form integrators page](https://mfem.org/lininteg/) and the [bilinear form integrators page](https://mfem.org/bilininteg/). Note that not all linear and bilinear forms that are available in MFEM have been implemented in MFEM-MOOSE. For a full list, check the kernels directory, `framework/src/mfem/kernels`. Should you wish to implement an integrator that is not yet available, please raise an issue in the MOOSE repository.
+
+If the integrator you wish to use is available, you can specify it in the `type` parameter simply by taking its MFEM integrator name, swapping the word `Integrator` for `Kernel`, and prepending MFEM to the beginning of its name. The table below shows a few examples of this naming convention:
+
+| MFEM name      | MFEM-MOOSE name      |
+| ------------- | ------------- |
+| `DiffusionIntegrator` | `MFEMDiffusionKernel` |
+| `CurlCurlIntegrator` | `MFEMCurlCurlKernel` |
+| `VectorDomainLFIntegrator` | `MFEMVectorDomainLFKernel` |
+
+Putting this together, our `Kernels` block might look as follows:
+
+!listing test/tests/mfem/kernels/diffusion.i block=/Kernels
+
+Now we set up boundary conditions. The full list of boundary conditions available may be found in the BCs directory, `framework/src/mfem/bcs`. Here, we choose scalar Dirichlet boundary conditions, which correspond to the [MFEMScalarDirichletBC.md] type.
```

Comment:
You don't have to address this comment here, this is for future reference.

We typically don't point users to a source directory in the documentation, i.e., `framework/src/mfem/bcs`, because users sometimes work with binary installs and don't always have direct access to the source tree. But I also see your dilemma here: there is no separate section listing the mfem BCs -- they are mangled with all the other framework BCs in the syntax documentation. @cticenhour Do you have any suggestion here?

Reply:
Similar problem goes to `framework/src/mfem/solvers` and `framework/src/mfem/outputs` below. I guess this works for now, just something to think about.

Reply:
I guess an alternative is to point to the corresponding section on the syntax page and say "go look for objects prefixed with MFEM" :)

Reply:
Source files and objects' syntax types don't always match, though.

Interestingly, @GiudGiud once suggested we make these optional packages (e.g., mfem, neml2) into modules (not git submodules, but moose physics modules). I haven't had a chance to thoroughly think through his suggestion at that time, but now I start to understand his arguments. That may naturally address issues like syntax listing. I don't know how much of effort it would be to adopt the "module" approach. But I think we'd better make a decision earlier than later, before this evolves into too much of a technical debt.

Reply:
> I guess an alternative is to point to the corresponding section on the syntax page and say "go look for objects prefixed with MFEM" :)

Done that, plus a link to the source page for completeness.
We still mention the tests directory, but that'll change soon in #30637 anyway. :)